examples/simulation/plot_simulated_sample_dataset.py

```diff
@@ -0,0 +1,181 @@
+"""
+=========================================
+Simulate raw data from the sample dataset
+=========================================
+
+This example illustrates how to generate source estimates and simulate raw data
+from the sample dataset using the :class:`mne.simulation.SourceSimulator' class.
+Once the raw data is simulated, generated source estimates are reconstructed 
+using Dynamic statistical parametric mapping (dSPM) inverse operator. 
+"""
+
+# Author: Ivana Kojcic <ivana.kojcic@gmail.com>
+#         Eric Larson <larson.eric.d@gmail.com>
+#         Kostiantyn Maksymenko <kostiantyn.maksymenko@gmail.com>
+#         Samuel Deslauriers-Gauthier <sam.deslauriers@gmail.com>
+
+# License: BSD (3-clause)
+
+import os.path as op 
+
+import numpy as np
+
+import mne
+from mne.datasets import sample
+
+print(__doc__)
+
+# To simulate the sample dataset, information of the sample subject needs to be
+# loaded. This step will download the data if it not already on your machine. 
+# Subjects directory is also set so it doesn't need to be given to functions.
+data_path = sample.data_path()
+subjects_dir = op.join(data_path, 'subjects')
+subject = 'sample'
+
+# First, we get an info structure from the sample subject.
+evoked_fname = op.join(data_path, 'MEG', subject, 'sample_audvis-ave.fif')
+info = mne.io.read_info(evoked_fname)
+tstep = 1 / info['sfreq']
+
+# To simulate sources, we also need a source space. It can be obtained from the
+# forward solution of the sample subject.
+meg_path = op.join(data_path, 'MEG', subject)
+fwd_fname = op.join(meg_path, 'sample_audvis-meg-eeg-oct-6-fwd.fif')
+fwd = mne.read_forward_solution(fwd_fname)
+src = fwd['src']
+
+################################################################################
+
+# Load the real raw data and corresponding events.
+raw = mne.io.read_raw_fif(op.join(meg_path, 'sample_audvis_raw.fif'))
+raw.set_eeg_reference(projection=True).crop(0, 60)  # for speed
```

Comment:
I would not use a raw object anywhere in the script.

You can file the events file in the sample dataset : see sample_audvis_raw-eve.fif
you can read the info with read_info function
you can find the noise cov also without recomputing it : sample_audvis-cov.fif

Reply:
Events read from sample_audvis_raw-eve.fif are != than the ones when I get with find_events(raw). I assume there's a good reason for that but I can't see why...